vimeo-networking/src/main/java/com/vimeo/networking/model/error/InvalidParameter.java

```diff
@@ -54,16 +57,27 @@ public InvalidParameter() {}
 
     public InvalidParameter(String field, ErrorCode errorCode, String developerMessage) {
         this.mField = field;
-        this.mErrorCode = errorCode;
+        setErrorCode(errorCode);
         this.mDeveloperMessage = developerMessage;
     }
 
-    public String getField() {
-        return mField;
+    private void setErrorCode(@NotNull ErrorCode errorCode) {
+        final String json = VimeoNetworkUtil.getGson().toJson(errorCode);
+        mRawErrorCode = json.replaceAll("^\"|\"$", "");
     }
 
+    @NotNull
     public ErrorCode getErrorCode() {
-        return mErrorCode == null ? ErrorCode.DEFAULT : mErrorCode;
+        final ErrorCode errorCode = VimeoNetworkUtil.getGson().fromJson(mRawErrorCode, ErrorCode.class);
```

Comment:
There is obviously a performance cost related to this that wasn't there before. Is there a reason why you can't just save the error code as a member when this class is initialized with it in the constructor or with setErrorCode with and return it here?

Reply:
It can be done, I will add in setters `setRawErrorCode` that deserialize the value and initialize the error code to reduce the performance code.